pkg/agent/endpoints/sdsv3/handler.go

```diff
@@ -244,29 +247,79 @@ func (h *Handler) buildResponse(versionInfo string, req *discovery_v3.DiscoveryR
 	}
 	returnAllEntries := len(names) == 0
 
+	// Use RootCA as default, but replace for SPIFFE auth when envoy version is at least v1.18.0
+	useRootCAValidator := true
```

Comment:
The builder interface doesn't seem very useful since we still check this boolean in most spots in order to set up the correct interface.

Perhaps, the builder interface looks like:
```
type validationContextBuilder interface {
    buildOne(resourceName, trustDomainID string) (*any.Any, error)
    buildAll(resourceName string, td string) (*any.Any, error)
}
```

Then that interface could be set once, up at top, once we determine if there is support for the SPIFFE Auth Extension. `buildOne` would be called in all cases except when the "ALL
 name is requested, which would use `buildAll`. The `buildAll` method for the "root ca" builder would fail.